examples/two-columns-layout/src/plugins/productsPlugin.tsx

```diff
@@ -0,0 +1,154 @@
+/** @jsx h */
+import {
+  AutocompletePlugin,
+  getAlgoliaResults,
+} from '@algolia/autocomplete-js';
+import { h } from 'preact';
+import { useCallback, useState } from 'preact/hooks';
+
+import { Blurhash } from '../components/Blurhash';
+import {
+  StarOutlineIcon,
+  StarFillIcon,
+  FavoriteFillIcon,
+  FavoriteOutlineIcon,
+} from '../components/Icons';
+import { ALGOLIA_PRODUCTS_INDEX_NAME } from '../constants';
+import { searchClient } from '../searchClient';
+import { ProductHit } from '../types';
+
+export interface ProductsPluginContext {
+  facetName: string;
+  facetValue: string;
+  query: string;
+}
+
+export const productsPlugin: AutocompletePlugin<ProductHit, {}> = {
+  getSources({ query }) {
+    if (!query) {
+      return [];
+    }
+
+    return [
+      {
+        sourceId: 'productsPlugin',
+        getItems() {
+          return getAlgoliaResults<ProductHit>({
+            searchClient,
+            queries: [
+              {
+                indexName: ALGOLIA_PRODUCTS_INDEX_NAME,
+                query,
+                params: {
+                  hitsPerPage: 4,
+                },
+              },
+            ],
+          });
+        },
+        onSelect({ setIsOpen }) {
+          setIsOpen(true);
+        },
+        templates: {
+          header() {
+            return <div className="aa-SourceHeaderTitle">Products</div>;
+          },
+          item({ item }) {
+            return <ProductItem hit={item} />;
+          },
+        },
+      },
+    ];
+  },
+};
+
+const formatPrice = (val: number) => {
+  return val.toFixed(2).toLocaleString().replace(/\./g, ',');
+};
+
+type ProductItemProps = {
+  hit: ProductHit;
+};
+
+const ProductItem = ({ hit }: ProductItemProps) => {
+  const [loaded, setLoaded] = useState(false);
+  const onLoad = useCallback(() => setLoaded(true), []);
+
+  const [favorite, setFavorite] = useState(false);
+  const onFavoriteClick = useCallback(
+    (event) => {
+      event.preventDefault();
+      event.stopPropagation();
+      setFavorite(!favorite);
+    },
+    [favorite]
+  );
+  const FavoriteIcon = favorite ? FavoriteFillIcon : FavoriteOutlineIcon;
+
+  const currentPrice = formatPrice(hit.price.value);
+  const discountedPrice = formatPrice(hit.price.discounted_value);
+  const priceCurrency = hit.price.currency === 'EUR' ? '€' : '$';
+
+  const rating = hit.reviews.rating;
+  const reviews = hit.reviews.count;
+
+  const stars = [];
+  for (let i = 0; i < 5; i++) {
+    const Star = i >= rating ? StarOutlineIcon : StarFillIcon;
+    stars.push(<li key={i}>{<Star className="aa-StarIcon" />}</li>);
+  }
+
+  return (
+    <a href="#" className="aa-ItemLink aa-ProductItem">
```

Comment:
If there's no `href`, should it be a link? If you change it to a `div` you can use `.aa-ItemWrapper` instead of `aa-ItemLink`.

Reply:
It should be a link in a real use-case. I used a link to have the pointer cursor and add a hover transition but maybe I should just use a `div` indeed. WDYT?

Reply:
Makes sense indeed if it's supposed to be a link. Maybe we could use `"https://example.org/"` which is designed for it?

Reply:
I could use the link you mentioned. Should we `preventDefault` in `onClick` so we don’t leave the example by mistake?